src/datastructure/KeyframeCollection.cpp

```diff
@@ -151,12 +164,32 @@ void KeyframeCollection::nextSerializationWithoutKeyframeImages()
     }
 }
 
+void KeyframeCollection::regularizeReferenceKeyframes()
```

Comment:
Do you use it? Is it useful? In theory, the code above should maintain a correct state. We shouldn't need this, no?